dp/cloud/go/services/dp/storage/amc_manager_test.go

```diff
@@ -0,0 +1,695 @@
+						WithSingleStepEnabled(true).
+						WithFullInstallationParam().
+						WithCbsdCategory("a").
+						Cbsd,
+					registered,
+					registered).
+				Details,
+		},
+	}, {
+		name: "test not get state without registration params",
+		inputCbsds: []*storage.DBCbsd{
+			b.NewDBCbsdBuilder().
+				Empty().
+				WithId(12).
+				WithNetworkId(someNetwork).
+				WithPreferences(15, []int64{3600, 3580, 3620}).
+				WithStateId(registeredId).
+				WithMinPower(1).
+				WithMaxPower(2).
+				WithNumberOfPorts(2).
+				WithDesiredStateId(registeredId).
+				Cbsd,
+		},
+		inputRequests: []*storage.DBRequest{},
+		inputGrants:   []*storage.DBGrant{},
+		expected:      []*storage.DetailedCbsd{},
+	}, {
+		name: "test not get state without eirp capabilities",
+		inputCbsds: []*storage.DBCbsd{
+			b.NewDBCbsdBuilder().
+				Empty().
+				WithId(13).
+				WithFccId(someFccId).
+				WithUserId(someUserId).
+				WithNetworkId(someNetwork).
+				WithCbsdId(someCbsdIdStr).
+				WithMinPower(1).
+				WithMaxPower(2).
+				WithNumberOfPorts(2).
+				WithSerialNumber(fmt.Sprintf(someSerialNumber+"%d", 13)).
+				WithPreferences(15, []int64{3600, 3580, 3620}).
+				WithStateId(registeredId).
+				WithDesiredStateId(registeredId).
+				Cbsd,
+		},
+		inputRequests: []*storage.DBRequest{},
+		inputGrants:   []*storage.DBGrant{},
+		expected:      []*storage.DetailedCbsd{},
+	}, {
+		name: "test not get state with single step enabled and no installation params",
+		inputCbsds: []*storage.DBCbsd{
+			b.NewDBCbsdBuilder().
+				WithNetworkId(someNetwork).
+				WithSerialNumber(fmt.Sprintf(someSerialNumber+"%d", 14)).
+				WithFccId(someFccId).
+				WithUserId(someUserId).
+				WithNetworkId(someNetwork).
+				WithCbsdId(someCbsdIdStr).
+				WithStateId(registeredId).
+				WithMinPower(1).
+				WithMaxPower(2).
+				WithNumberOfPorts(2).
+				WithDesiredStateId(registeredId).
+				WithSingleStepEnabled(true).
+				WithCbsdCategory("a").
+				Cbsd,
+		},
+		inputRequests: []*storage.DBRequest{},
+		inputGrants:   []*storage.DBGrant{},
+		expected:      []*storage.DetailedCbsd{},
+	}, {
+		name: "test not get state with single step enabled, category A and outdoor",
+		inputCbsds: []*storage.DBCbsd{
+			b.NewDBCbsdBuilder().
+				WithNetworkId(someNetwork).
+				WithSerialNumber(fmt.Sprintf(someSerialNumber+"%d", 14)).
+				WithFccId(someFccId).
+				WithUserId(someUserId).
+				WithNetworkId(someNetwork).
+				WithCbsdId(someCbsdIdStr).
+				WithMinPower(1).
+				WithMaxPower(2).
+				WithNumberOfPorts(2).
+				WithStateId(registeredId).
+				WithDesiredStateId(registeredId).
+				WithSingleStepEnabled(true).
+				WithCbsdCategory("a").
+				WithFullInstallationParam().
+				WithIndoorDeployment(false).
+				Cbsd,
+		},
+		inputRequests: []*storage.DBRequest{},
+		inputGrants:   []*storage.DBGrant{},
+		expected:      []*storage.DetailedCbsd{},
+	}}
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			for _, c := range tc.inputCbsds {
```

Comment:
You can do this to keep setup in one transaction:
```
testCases := []struct {
		name          string
		input []db.Model // keep cbsds grants, requests here
		expected      []*storage.DetailedCbsd
	}{{
````

Reply:
done